### PR TITLE
feat: farms pools data

### PR DESCRIPTION
**Pull request by ChefJerry**
---

## PR-Codex overview
This PR adds support for a new protocol version (v4) in farms configuration and introduces new types and functions related to v4 farms.

### Detailed summary
- Added `Protocol` type 'v4bin'
- Updated `FarmBaseConfig` with 'v4bin' protocol
- Added `PoolInfo` type for v4 farms
- Added `ChainIdAddressKey` type
- Added `supportedChainIdV4` in `const.ts`
- Updated functions and types for v4 farms in various files

> The following files were skipped due to too many changes: `apps/web/src/state/farmsV4/state/farmPoolsAtom.ts`, `apps/web/src/state/info/api/schema.d.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/state/farmsV4/state/utils.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  data:
+    | paths['/cached/pools/farming']['get']['responses']['200']['content']['application/json']
+    | paths['/cached/pools/list']['get']['responses']['200']['content']['application/json'],
+): PoolInfo[] => {
+  return data.map((pool) => {
+    return {
+      chainId: pool.chainId,
+      lpAddress: safeGetAddress(pool.id)!,
+      protocol: pool.protocol,
+      token0: new Token(
```
Comment (chef-eric) [inline apps/web/src/state/farmsV4/state/utils.ts:16]: can Token assign to Currency type?

File: apps/web/src/state/farmsV4/state/utils.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        pool.chainId,
+        safeGetAddress(pool.token1.id)!,
+        pool.token1.decimals,
+        pool.token1.symbol,
+        pool.token1.name,
+      ),
+      lpApr: pool.apr24h as `${number}`,
+      tvlUsd: pool.tvlUSD as `${number}`,
+      vol24hUsd: pool.volumeUSD24h as `${number}`,
+      feeTier: BigInt(pool.feeTier),
```
Comment (chef-eric) [inline apps/web/src/state/farmsV4/state/utils.ts:33]: the FeeTier UIkit props:fee is number type, should we turn it to bigint?

**Review by chef-eric — APPROVED** [review]

File: apps/web/src/state/farmsV4/state/utils.ts (hunk 1 above)
Comment (ChefJerry) [reply apps/web/src/state/farmsV4/state/utils.ts:16]: see https://github.com/pancakeswap/pancake-frontend/blob/254e13b7d24319c78a21788990fe97381fc15970/packages/swap-sdk-core/src/currency.ts#L4

File: apps/web/src/state/farmsV4/state/utils.ts (hunk 2 above)
Comment (ChefJerry) [reply apps/web/src/state/farmsV4/state/utils.ts:33]: ok, let's use number

**Pull request merged**